File: front/src/app/models/light-dto.ts
import { LngLatLike } from 'mapbox-gl';
import { TimePeriodSetting } from './time-period';

export interface LightDto {
  address: string | null;
  note: string | null;
  disableAfterSeconds: number | null;
  proximityActivationRadius: number | null;
  brightnessConfig: Array<TimePeriodSetting> | null;
  position: LngLatLike;
}

File: backend/src/main/java/com/greencity/backend/model/dto/LightDto.java
package com.greencity.backend.model.dto;

import jakarta.validation.Valid;

import java.util.List;

/**
 * @author Anton Gorokh
 */
public record LightDto(
		@Valid GeoPosition position,
		String address,
		String note,
		Integer disableAfterSeconds,
		Double proximityActivationRadius,
		List<TimePeriodSetting> brightnessConfig
) {
}
